app/services/ping_admins.rb

```diff
@@ -9,7 +9,7 @@ def self.call(*args)
   end
 
   def call
-    return unless user && Rails.env.production?
```

Comment:
the issue with removing this is that in development mode this is what happens: 

```ruby
[2] pry(main)> PingAdmins.call(User.last)
  User Load (24.5ms)  SELECT  "users".* FROM "users" ORDER BY "users"."id" DESC LIMIT $1  [["LIMIT", 1]]
NoMethodError: undefined method `request_uri' for #<URI::Generic Optional>
from /Users/rhymes/.rvm/gems/ruby-2.6.3/gems/slack-notifier-2.3.2/lib/slack-notifier/util/http_client.rb:41:in `request_obj'
```

the same calls does nothing in test mode. 

I think a better approach would be to remove this yes but also to modify the initializer to make sure that in development the `create_test_channel_notifier` is only called when the variable `SLACK_WEBHOOK_URL` is used, otherwise to initialize it with the no-op notifier.

Unfortunately there's also a bug in the `Envfile` which contains the default values of the variables, they should be:

```ruby
variable :SLACK_CHANNEL, :String, default: ""
variable :SLACK_WEBHOOK_URL, :String, default: ""
```

and right now are

```ruby
variable :SLACK_CHANNEL, :String, default: "Optional"
variable :SLACK_WEBHOOK_URL, :String, default: "Optional"
```

this way in the initializer you can check if `SLACK_WEBHOOK_URL` is set to anything and use it to initialize the slack notifier

Reply:
Can you take a look agin and see if it works now?

Reply:
@rhymes Do we need to mention about setting SLACK_WEBHOOK_URL in the docs? Not sure if it is already documented. Also, `SLACK_CHANNEL` is only used in the production environment. What do you think about that?

Reply:
> Because returning stubbed notifier in the development will not raise an error but it will also not send an alert to actual Slack

I think that's an okay situation. The `Envfile` is setup as to require the minimum amount of external services needed for the app to function, pinging admins on your local environment is not a core functionality. If the user wants to test that they will obviously have to setup their own slack webhook and channel. So I think the default implemented by @AllanKlaus is safe enough

I'm +1 with the idea of having the `SLACK_CHANNEL` configurable in development mode as well.

Regarding the documentation, we might want to setup a sort of "optional services" page in the backend guide that documents how to setup all the optional 3rd party services in the `Envfile`, why not! It can be kickstarted by Slack

I think they should be done in separate PRs though (both the slack channel refactor and the documentation).

Reply:
@AllanKlaus it works!